src/foam/nanos/crunch/CapabilityAvailabilityDAO.js

```diff
@@ -62,19 +67,38 @@ foam.CLASS({
     {
       name: 'select_',
       javaCode: `
-        AuthService auth = (AuthService) x.get("auth");
-        Sink s = sink != null ? sink : new ArraySink();
-        ProxySink proxy = new ProxySink(x, s) {
-          public void put(Object o, Detachable d) {
-            Capability capability = (Capability) o;
-            if ( capability.getAvailabilityPredicate().f(x) || auth.check(x, AVAILABILITY_PERMISSION + capability.getId()) ) {
-              getDelegate().put(capability, d);
-            }
+        return getDelegate().select_(x, sink, skip, limit, order, augmentPredicate(x, predicate));
+      `
+    },
+    {
+      name: 'augmentPredicate',
```

Comment:
Is this method needed? Couldn't you just AND() this predicate and the supplied 'predicate' together?

Reply:
yes, it's needed since unlike the regular predicate we need both x and the capability obj.